libs/cart/src/drivers/magento/cart.service.ts

```diff
@@ -53,14 +53,17 @@ export class DaffMagentoCartService implements DaffCartServiceInterface<DaffCart
   clear(cartId: string): Observable<Partial<DaffCart>> {
     return this.cartItemDriver.list(cartId).pipe(
       switchMap(items =>
-        // make the delete requests in parallel and collect them into a single observable
-        zip(...items.map(item =>
+				// make the delete requests in parallel and collect them into a single observable
+				// return null if there are no items in the cart
+        items.length ? zip(...items.map(item =>
           this.cartItemDriver.delete(cartId, item.item_id)
-        ))
+        )) : of(null)
       ),
       // find the most up to date delete response by looking for one with no items
-      // it might be necessary to make a get request instead if delete does not return needed fields
-      map(updatedCarts => updatedCarts.find(cart => cart.items.length === 0))
+      // make a get request if there were no items in the cart
+			switchMap(updatedCarts => !!updatedCarts ? 
```

Comment:
I think `!!` is unnecessary if you don't actually need an explicit boolean. For the purposes of a conditional it should be sufficient to do `updatedCarts ?`.

Reply:
changed